File: PSET1/mario/less/mario.c
#import <stdio.h>
#import <cs50.h>

int main(void)
{
    // create variables
    int height;
    int spaces;
    int blocks;

    // get input > 0 and <= 23
    do
    {
        printf("Pyramid height :\n");
        height = get_int();
    }
    while (height < 0 || height > 23);

    // iterate over pyramid height
    for (int i = 0; i < height; i++)
    {
        // for each row: print spaces
        for (spaces = height - 1; spaces > i; spaces--)
        {
            printf(" ");
        }
        // for each row: print blocks
        for (blocks = 0; blocks <= i + 1; blocks++)
        {
            printf("#");
        }
        // go to next line
        printf("\n");
    }
}

File: PSET6/mario.py
# import get_int from cs50 library
from cs50 import get_int

# prompt user for int between 0 and 23
while True:
    print("Pyramid height")
    height = get_int()
    if(height >= 0 and height <= 23):
        break

# for each row
for i in range(height):

    # print spaces
    spaces = height - 1
    while spaces > i:
        print(" ", end="")
        spaces -= 1

    # print blocks first pyramid
    blocks = 0
    while blocks <= i:
        print("#", end="")
        blocks += 1

    # print spaces between 2 pyramids
    print("  ", end="")

    # print block second pyramid
    blocks = 0
    while blocks <= i:
        print("#", end="")
        blocks += 1

    # new line after each row
    print()
